src/EditorFeatures/Text/Extensions.cs

```diff
@@ -37,7 +40,7 @@ public static SourceText AsText(this ITextSnapshot textSnapshot)
             => SnapshotSourceText.From(textSnapshot);
 
         internal static SourceText AsRoslynText(this ITextSnapshot textSnapshot, Encoding encoding)
-            => new SnapshotSourceText.ClosedSnapshotSourceText(textSnapshot, encoding);
+            => new SnapshotSourceText.ClosedSnapshotSourceText(((ITextSnapshot2)textSnapshot).TextImage, encoding);
```

Comment:
this is only called from closed file. open file went through different code path. closed file has different type since it made memory investigation easier.